src/components/smartSearch/filters/TimeFrame.tsx

```diff
@@ -0,0 +1,120 @@
+import DateFnsUtils from '@date-io/date-fns';
+import { FormattedMessage as Msg } from 'react-intl';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { MenuItem, Typography } from '@material-ui/core';
+import { useEffect, useState } from 'react';
+
+import { getNewDateWithOffset } from 'utils/dateUtils';
+import { getTimeFrame } from '../utils';
+import StyledDatePicker from '../inputs/StyledDatePicker';
+import StyledNumberInput from '../inputs/StyledNumberInput';
+import StyledSelect from '../inputs/StyledSelect';
+import { TIME_FRAME } from 'types/smartSearch';
+
+interface TimeFrameProps {
+    onChange: (range: {after?: string; before?: string}) => void;
+    filterConfig?: {after?: string; before?: string};
+}
+
+const TimeFrame = ({ onChange, filterConfig }: TimeFrameProps): JSX.Element => {
+    const [selected, setSelected] = useState<TIME_FRAME>(TIME_FRAME.EVER);
+    const today = new Date();
+    const [before, setBefore] = useState(today);
+    const [after, setAfter] = useState(getNewDateWithOffset(today, -30));
+    const [numDays, setNumDays] = useState(30);
+
+    useEffect(() => {
+        if (filterConfig) {
+            const timeFrame = getTimeFrame(filterConfig);
+            const { after, before } = filterConfig;
+            setSelected(timeFrame);
+            if (timeFrame === TIME_FRAME.LAST_FEW_DAYS && after) {
+                setNumDays(+after.substring(1, after.length - 1));
+                return;
+            }
+            else if (timeFrame === TIME_FRAME.BETWEEN && after && before) {
+                setAfter(new Date(after));
+                setBefore(new Date(before));
+            }
+            else if (timeFrame === TIME_FRAME.AFTER_DATE && after) {
+                setAfter(new Date(after));
+            }
+            else if (timeFrame === TIME_FRAME.BEFORE_DATE && before) {
+                setBefore(new Date(before));
+            }
+        }
+    }, []);// eslint-disable-line react-hooks/exhaustive-deps
+    useEffect(() => {
+        if (selected ===  TIME_FRAME.EVER) {
+            onChange({});
```

Comment:
Invoking the `onChange()` callback in a `useEffect` instead of doing it directly in the event handlers for the various changes felt weird to me, but I tried to write a nicer solution without it and couldn't without making significant changes to your structure (and I'm not sure I would end up in a better place). 🙄 

Do you have any thoughts, @djbusstop?

Reply:
I guess it could be argued that the `onChange()` is an _effect_ of changing the internal state, so lets leave this for now. :)